gems/smithy-client/lib/smithy-client/pageable_response.rb

```diff
@@ -133,6 +133,14 @@ def next_page_params(params)
         new_params = context.params.except(*prev_tokens)
         new_params.merge!(@paginator.next_tokens(data).merge(params))
       end
+
+      def with_metric(&block)
+        Thread.current[:aws_sdk_core_user_agent_metric] ||= []
+        Thread.current[:aws_sdk_core_user_agent_metric] << 'C'
+        block.call
+      ensure
+        Thread.current[:aws_sdk_core_user_agent_metric].pop
+      end
```

Comment:
Should we put a TODO to revisit this approach? Just want to make sure that this does not go amiss